compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5587,6 +5599,15 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
                 return new BStreamType(TypeTags.STREAM, selectType, errorType, null);
             } else if (queryExpr.isTable) {
                 actualType = getQueryTableType(queryExpr, selectType);
+            } else if (queryExpr.isMap) {
+                BType mapConstraintType = getTypeOfTypeParameter(selectType,
+                        queryExpr.getSelectClause().expression.pos);
+                if (mapConstraintType == symTable.semanticError) {
+                    actualType = symTable.semanticError;
```

Comment:
It is not needed for this assignment because actualType is already symTable.semanticError.